agent/firewall.go

```diff
@@ -407,12 +407,14 @@ func provisionFirewallRules(netif NetIf, agent *Agent) error {
 	if err != nil {
 		return err
 	}
-	for chain := range missingChains {
+	for _, chain := range missingChains {
 		if err := fw.CreateRules(chain); err != nil {
 			return err
 		}
-		if err := fw.CreateU32Rules(chain); err != nil {
-			return err
+		if chain == forwardChainIndex {
```

Comment:
We could add a comment here that communication between endpoints always takes place via the FORWARD chain, since INPUT and OUTPUT are exclusively used for communication with the host. Thus, the u32 rule only needs to be added to FORWARD.